src/OrchardCore.Modules/OrchardCore.Placements/Startup.cs

```diff
@@ -0,0 +1,81 @@
+using OrchardCore.Placements.Deployment;
+using OrchardCore.Placements.Recipes;
+using OrchardCore.Placements.Services;
+using OrchardCore.Placements.Settings;
+using OrchardCore.Recipes;
+using OrchardCore.Security.Permissions;
+
+namespace OrchardCore.Placements
+{
+    public class Startup : StartupBase
+    {
+        private readonly AdminOptions _adminOptions;
+
+        public Startup(IOptions<AdminOptions> adminOptions)
+        {
+            _adminOptions = adminOptions.Value;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IPermissionProvider, Permissions>();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+
+            services.AddScoped<IPlacementsManager, PlacementsManager>();
+            services.AddScoped<IShapePlacementProvider, PlacementProvider>();
+
+            // Shortcuts in settings
+            services.AddScoped<IContentPartDefinitionDisplayDriver, PlacementContentPartDefinitionDriver>();
+            services.AddScoped<IContentTypePartDefinitionDisplayDriver, PlacementContentTypePartDefinitionDriver>();
+            services.AddScoped<IContentPartFieldDefinitionDisplayDriver, PlacementContentPartFieldDefinitionDisplayDriver>();
+
+            // Deployments
+            services.AddTransient<IDeploymentSource, PlacementsDeploymentSource>();
+            services.AddSingleton<IDeploymentStepFactory>(new DeploymentStepFactory<PlacementsDeploymentStep>());
+            services.AddScoped<IDisplayDriver<DeploymentStep>, PlacementsDeploymentStepDriver>();
+
+            // Recipies
+            services.AddRecipeExecutionStep<PlacementStep>();
+        }
+
+        public override void Configure(IApplicationBuilder builder, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var templateControllerName = typeof(AdminController).ControllerName();
+
+            routes.MapAreaControllerRoute(
+                name: "Placements.Index",
+                areaName: "OrchardCore.Placements",
+                pattern: _adminOptions.AdminUrlPrefix + "/Placements",
+                defaults: new { controller = templateControllerName, action = nameof(AdminController.Index) }
+            );
+
+            routes.MapAreaControllerRoute(
+                name: "Placements.Create",
+                areaName: "OrchardCore.Placements",
+                pattern: _adminOptions.AdminUrlPrefix + "/Placements/Create",
+                defaults: new { controller = templateControllerName, action = nameof(AdminController.Create) }
+            );
+
+            routes.MapAreaControllerRoute(
+                name: "Placements.Edit",
+                areaName: "OrchardCore.Placements",
+                pattern: _adminOptions.AdminUrlPrefix + "/Placements/Edit/{shapeType}",
+                defaults: new { controller = templateControllerName, action = nameof(AdminController.Edit) }
+            );
```

Comment:
No routing for `Remove` ?

Should it be `Remove` or `Delete` - we do have some inconsistency about that